os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -764,12 +764,37 @@ static int i2s_rxdma_prep(struct amebasmart_i2s_s *priv, struct amebasmart_buffe
 	apb->nbytes = 0;
 	apb->curbyte = 0;
 
-	priv->i2s_rx_buf = apb->samp;
 	i2s_set_dma_buffer(&priv->i2s_object, NULL, (char *)priv->i2s_rx_buf, I2S_DMA_PAGE_NUM, I2S_DMA_PAGE_SIZE);
 
 	return 0;
 }
 
+static int amebasmart_i2s_rx(struct amebasmart_i2s_s *priv, struct amebasmart_buffer_s *bfcontainer) {
+
+	int ret = OK;
+	int rx_size;
+
+	rx_size = I2S_DMA_PAGE_SIZE; /* Track current byte size to increment by */
+	struct ap_buffer_s *apb;
+
+	if (NULL != bfcontainer && NULL != bfcontainer->apb) {
+
+		apb = bfcontainer->apb;
+		/* Add the container to the list of active DMAs */
+		sq_addlast((sq_entry_t *)bfcontainer, &priv->rx.act);
+
+		priv->apb_rx = apb; /* Reference current TX apb in our I2S struct */
+							/* Start sending first page, after that the txdma callback will be called in the tx irq handler */
+	}
+	i2s_recv_page(&priv->i2s_object, (u8 *) &priv->apb_rx->samp[priv->apb_rx->curbyte], I2S_DMA_PAGE_SIZE);
```

Comment:
I think this statement should be inside the if condition, could you please check?

Reply:
Understood, datawidth modification and parameter should be in if condition